### DISCUSSION TITLE
Distinguish whether devices belong to the same network, obtaining Joiner UDP port and steering data

### DISCUSSION THREAD

**Discussion by ferre111** [openthread/openthread#9329] [Q&A] [answered]
Hello!

I have two questions:
1. Is there a way to check if the devices returned by the `scan` or `discover` commands belong to the same network? Or maybe there is separate command for this purpose?
2. How can I obtain data such as `steering data` or `Joiner UDP port`? I see that previously it was possible by using the dBus interface `io.openthread.BorderRouter`, but now this is not the case because Thread Specification v1.2.1 removes support for Thread Beacons payload. Is there a chance to obtain this data using `Discover` operation and some other dBus method?

**Comment by jwhui** [comment]
> 1. Is there a way to check if the devices returned by the `scan` or `discover` commands belong to the same network? Or maybe there is separate command for this purpose?

The Extended PAN ID is a 64-bit unique identifier for the network.

> 2. How can I obtain data such as `steering data` or `Joiner UDP port`? I see that previously it was possible by using the dBus interface `io.openthread.BorderRouter`, but now this is not the case because Thread Specification v1.2.1 removes support for Thread Beacons payload. Is there a chance to obtain this data using `Discover` operation and some other dBus method?

The MLE Discover operation should return equivalent information in the [`otActiveScanResult()`](https://openthread.io/reference/struct/ot-active-scan-result) as part of the callback.

**Reply by ferre111** [reply]
Thank you for the answer. 

> The Extended PAN ID is a 64-bit unique identifier for the network.

But I was able to create separate networks with the same Extended PANID, both devices can see each other and both were the leader in their network.

> The MLE Discover operation should return equivalent information in the [otActiveScanResult()](https://openthread.io/reference/struct/ot-active-scan-result) as part of the callback.

But currently it is not possible to get this information via CLI or dbus?

**Reply by jwhui** [reply, marked as answer]
> But I was able to create separate networks with the same Extended PANID, both devices can see each other and both were the leader in their network.

If they have the same Network Key, they will be able to participate in the same Thread network. When creating a new Thread network, it is required to randomly generate a new Extended PAN ID and Network Key.

> But currently it is not possible to get this information via CLI or dbus?

Not yet. Feel free to submit a pull request if you would like!

**Reply by ferre111** [reply]
Now everything is clear, thanks!
